[PREG-183] Remove async mode from pregel (#1221)
arangodb/docs: 3.11/release-notes-upgrading-changes311.md

From v3.11.0 onwards, this option is deprecated, and setting it to a value
different than the value of `--agency.size` leads to a startup error.

### Pregel options

> The `asyinc` option used in algorithms which support an asynchronous mode
> to run without synchronized global iterations has been removed.

Reviewer: ```suggestion
The `async` option has been removed. Some algorithms supported an asynchronous mode
to run without synchronized global iterations. This is no longer supported.
```
